src/Kestrel.Core/Internal/Infrastructure/HttpCharacters.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure
+{
+    internal static class HttpCharacters
+    {
+        public static readonly int TableSize = 128;
+        public static readonly ReadOnlyMemory<bool> Authority;
+        public static readonly ReadOnlyMemory<bool> Token;
+        public static readonly ReadOnlyMemory<bool> Host;
+        public static readonly ReadOnlyMemory<bool> FieldValue;
+
+        static HttpCharacters()
```

Comment:
Dump static constructors; move all static constants to a `Constants` class where the statics are initalized inline so the class is marked `.beforefieldinit`

e.g.
```csharp
public static readonly ReadOnlyMemory<bool> Authority = InitalizeAuthority();
```

Then access a single one of the readonly statics; outside of the fast-path, so they all initalize with the initialization checks embedded in a non-hot method.

The static constructor will cause the thread-safe initialisation check to be added to every access. 

`.beforefieldinit` will cause the thread-safe initialisation check to be added to the first access.

Reply:
Does this have to be in a different `Constants` class? Would it not work to initialize the statics inline in this class?

Reply:
Yes, but there several different sets of readonly statics; so it would be easiest to collate them all, then only needs a single access to initalize them all, rather than one access per class.

Reply:
I'm not too fond of creating a separate constants class so I just kept all of the statics here.